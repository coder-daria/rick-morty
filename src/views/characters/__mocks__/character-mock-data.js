export const LOCATION_NAME = 'Earth';

export const CHARACTER_NOT_FORMATED_MOCKED_DETAILS = {
  location: {
    name: LOCATION_NAME,
  },
  origin: {
    name: LOCATION_NAME,
  },
};

export const CHARACTER_FORMATED_MOCKED_DETAILS = {
  location: LOCATION_NAME,
  origin: LOCATION_NAME,
};

export const CHARACTER_DETAILED_MOCKED_DATA = {
  created: '11/25/2020, 4:59:08 PM',
  episode: [],
  gender: 'Male',
  image: '',
  name: 'Rick Sanchez',
  species: 'Alien',
  status: 'Alive',
  type: '',
};
