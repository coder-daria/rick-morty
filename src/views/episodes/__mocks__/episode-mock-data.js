export const EPISODE_NOT_FORMATED_MOCKED_DETAILS = {
  air_date: '11/25/2020, 4:59:08 PM',
};

export const EPISODE_FORMATED_MOCKED_DETAILS = {
  'air date': '11/25/2020, 4:59:08 PM',
};

export const EPISODE_DETAILS_MOCKED_DATA = {
  name: 'Sunny day',
  episode: '0E12',
  created: '11/25/2019, 4:59:08 PM',
  characters: [{ name: 'Tom' }],
};
