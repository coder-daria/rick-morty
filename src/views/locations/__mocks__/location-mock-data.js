export const LOCATION_DETAILS_MOCKED_DATA = {
  name: 'Earth',
  created: '11/25/2019',
  dimension: '',
  type: '',
  residents: [{ name: 'Tom' }],
};
