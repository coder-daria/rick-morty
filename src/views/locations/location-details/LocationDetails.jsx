import React, { useMemo } from 'react';
import { instanceOf, shape, string } from 'prop-types';
import { Col } from 'antd';

import { renderCharactersNames } from '../../../common/utils/characters';

import LocationModel from '../model/LocationModel';

import { StyledRow } from './LocationDetails.styles';

const NO_DATA_PLACEHOLDER = '-';

function LocationDetails({ location }) {
  const locationDetails = useMemo(() => LocationModel(location), [location]);

  return (
    <div>
      {Object.keys(locationDetails).map(item => (
        <StyledRow key={item}>
          <Col span={12}>{item}</Col>
          <Col span={12}>
            {item === 'residents'
              ? renderCharactersNames(locationDetails[item])
              : locationDetails[item] || NO_DATA_PLACEHOLDER}
          </Col>
        </StyledRow>
      ))}
    </div>
  );
}

LocationDetails.propTypes = {
  location: shape({
    id: string,
    created: string,
    dimension: string,
    name: string,
    type: string,
    residents: instanceOf(Array),
  }).isRequired,
};

export default LocationDetails;
