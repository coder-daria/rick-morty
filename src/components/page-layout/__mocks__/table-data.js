import { CHARACTER_NOT_FORMATED_MOCKED_DETAILS } from '../../../views/characters/__mocks__/character-mock-data';

export const PAGE_INFO = {
  pages: 1,
  count: 1,
  next: null,
  prev: null,
};

export const TABLE_DATA = {
  columns: [
    {
      title: 'Name',
      dataIndex: 'name',
      key: 'name',
    },
    {
      title: 'Species',
      dataIndex: 'species',
      key: 'species',
    },
    {
      title: 'Origin',
      dataIndex: 'origin',
      key: 'origin',
    },
    {
      title: 'Location',
      dataIndex: 'location',
      key: 'location',
    },
  ],
  loading: false,
  pageInfo: PAGE_INFO,
  tableData: [
    {
      id: 1,
      name: 'Rick Sanchez',
      species: 'Alien',
      ...CHARACTER_NOT_FORMATED_MOCKED_DETAILS,
    },
  ],
  toggleDrawer: jest.fn(),
  setCurrentPage: jest.fn(),
};
