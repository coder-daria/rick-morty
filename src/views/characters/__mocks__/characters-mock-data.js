import { PAGE_INFO } from '../../../components/page-layout/__mocks__/table-data';
import { CHARACTER_NOT_FORMATED_MOCKED_DETAILS } from './character-mock-data';

const CHARACTER_TABLE_MOCKED_DATA = {
  id: 1,
  name: 'Rick Sanchez',
  species: 'Alien',
  ...CHARACTER_NOT_FORMATED_MOCKED_DETAILS,
};

export const CHARACTERS_MOCKED_DATA = {
  characters: {
    results: [CHARACTER_TABLE_MOCKED_DATA],
    info: PAGE_INFO,
  },
};
