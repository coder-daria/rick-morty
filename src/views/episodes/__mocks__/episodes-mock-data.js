import { PAGE_INFO } from '../../../components/page-layout/__mocks__/table-data';

const EPISODE_TABLE_MOCKED_DATA = {
  characters: [{ name: 'Tom' }],
  episode: '0E12',
  id: 1,
  name: 'Sunny day',
};

export const EPISODES_MOCKED_DATA = {
  episodes: {
    results: [EPISODE_TABLE_MOCKED_DATA],
    info: PAGE_INFO,
  },
};
