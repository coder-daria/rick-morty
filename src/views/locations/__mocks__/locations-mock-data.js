import { PAGE_INFO } from '../../../components/page-layout/__mocks__/table-data';

const LOCATION_TABLE_MOCKED_DATA = {
  id: 1,
  created: '11/25/2019, 4:59:08 PM',
  dimension: '',
  name: 'Sunny day',
  type: '',
  residents: [{ name: 'Tom' }],
};

export const LOCATIONS_MOCKED_DATA = {
  locations: {
    results: [LOCATION_TABLE_MOCKED_DATA],
    info: PAGE_INFO,
  },
};
