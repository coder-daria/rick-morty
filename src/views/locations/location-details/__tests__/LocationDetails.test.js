import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { LOCATION_DETAILS_MOCKED_DATA } from '../../__mocks__/location-mock-data';

import LocationDetails from '../LocationDetails';

const setup = () =>
  shallow(<LocationDetails location={LOCATION_DETAILS_MOCKED_DATA} />);

describe('<LocationDetails />', () => {
  it('should match the snapshot', () => {
    const wrapper = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
