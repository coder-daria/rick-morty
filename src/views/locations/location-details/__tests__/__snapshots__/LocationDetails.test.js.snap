// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationDetails /> should match the snapshot 1`] = `
<div>
  <Styled(Row)
    key="name"
  >
    <Col
      span={12}
    >
      name
    </Col>
    <Col
      span={12}
    >
      Earth
    </Col>
  </Styled(Row)>
  <Styled(Row)
    key="created"
  >
    <Col
      span={12}
    >
      created
    </Col>
    <Col
      span={12}
    >
      11/25/2019
    </Col>
  </Styled(Row)>
  <Styled(Row)
    key="dimension"
  >
    <Col
      span={12}
    >
      dimension
    </Col>
    <Col
      span={12}
    >
      -
    </Col>
  </Styled(Row)>
  <Styled(Row)
    key="type"
  >
    <Col
      span={12}
    >
      type
    </Col>
    <Col
      span={12}
    >
      -
    </Col>
  </Styled(Row)>
  <Styled(Row)
    key="residents"
  >
    <Col
      span={12}
    >
      residents
    </Col>
    <Col
      span={12}
    >
      <div
        key="Tom0"
      >
        Tom
      </div>
    </Col>
  </Styled(Row)>
</div>
`;
